R/load.R

```diff
@@ -652,66 +653,105 @@ loadDistances <- function(input = "distances.csv", spatial) {
   invalid.dist <- TRUE
   if (is.character(input)) {
     if (file.exists(input)) {
-      appendTo(c("Screen", "Report"), paste0("M: File '", input, "' found, activating speed calculations."))
-      dist.mat <- read.csv(input, row.names = 1, check.names = FALSE)
-      if (ncol(dist.mat) == 1)
-        warning("Only one column was identified in '", input, "'. If this seems wrong, please make sure that the values are separated using commas.", immediate. = TRUE, call. = FALSE)
+      appendTo(c("Screen", 
+                 "Report"), 
+               paste0("M: File '", 
+                      input, "' found, activating speed calculations."))
+      dist.mat <- as.matrix(read.csv(input, row.names = 1, check.names = FALSE))
+      if (ncol(dist.mat) == 1){
+        warning(paste0("Only one column was identified in '", 
+                input, "'. If this seems wrong,", 
+                " please make sure that the values are separated using commas."), 
+                immediate. = TRUE, 
+                call. = FALSE)}
     } else {
       dist.mat <- NULL
     }
-  } else {
-    # Ensure the input is a data frame, 
+  } else{
+    dist.mat <- as.matrix(input)
+  } 
+    # Ensure the input is a matrix, 
     # in case the user provided the data in a non-base format
-    # Some of the code downstream won't work with a matrix,  
-    # so the input can't be converted to a matrix right away
-    dist.mat <- as.data.frame(input)
-  }
+  
 
   if (!is.null(dist.mat)) {
     rownames(dist.mat) <- gsub(" ", "_", rownames(dist.mat))
     colnames(dist.mat) <- gsub(" ", "_", colnames(dist.mat))
     invalid.dist <- FALSE
     if (nrow(dist.mat) != ncol(dist.mat)){
-      appendTo(c("Screen", "Report", "Warning"), "The distances matrix appears to be missing data (ncol != nrow). Deactivating speed calculations to avoid function failure.")
+      appendTo(c("Screen", 
+                 "Report", 
+                 "Warning"), 
+               paste0("The distances matrix appears to be missing data", 
+                      " (ncol != nrow). Deactivating speed calculations", 
+                      " to avoid function failure."))
       invalid.dist <- TRUE
     }
-    if (!invalid.dist && any(link <- is.na(match(rownames(dist.mat), colnames(dist.mat))))) {
-      appendTo(c("Screen", "Report", "Warning"), "The column and row names in the distances matrix do not match each other. Deactivating speed calculations to avoid function failure.")
-      message(paste0("   Row names missing in the columns: '", paste(rownames(dist.mat)[link], collapse = "', '"), "'."))
-      if (any(link <- is.na(match(colnames(dist.mat), rownames(dist.mat)))))
-        message(paste0("   Column names missing in the rows: '", paste(colnames(dist.mat)[link], collapse = "', '"), "'."))
+    if (!invalid.dist && any(link <- is.na(match(rownames(dist.mat), 
+                                                 colnames(dist.mat))))) {
```

Comment:
let's take the chance to break this down, I don't like that I used to define objects inside if statements.

Suggestion for the whole block:

```r
    if (!invalid.dist) {
      row_link <- is.na(match(rownames(dist.mat), colnames(dist.mat)))
      col_link <- is.na(match(colnames(dist.mat), rownames(dist.mat)))
      if (any(row_link) | any(col_link)) {
         appendTo(c("Screen",
                 "Report",
                 "Warning"),
               paste0("The column and row names in the distances matrix do not",
                      " match each other. Deactivating speed calculations to",
                      " avoid function failure."))
        if (any(row_link)) {
          message(paste0("   Row names missing in the columns: '",
                         paste(rownames(dist.mat)[row_link],
                               collapse = "', '"), "'."))
        }
        if (any(col_link)) {
          message(paste0("   Column names missing in the rows: '",
                         paste(colnames(dist.mat)[col_link],
                               collapse = "', '"), "'."))
        }
      }
    }
```